services/worker-manager/src/provisioner.js

```diff
@@ -1,25 +1,34 @@
+const taskcluster = require('taskcluster-client');
 const Iterate = require('taskcluster-lib-iterate');
+const {consume} = require('taskcluster-lib-pulse');
 
 /**
  * Run all provisioning logic
  */
 class Provisioner {
-  constructor({queue, provisionerId, providers, iterateConf, WorkerType, monitor, notify}) {
-    this.queue = queue;
+  constructor({provisionerId, providers, iterateConf, WorkerType, monitor, notify, pulseClient, reference, rootUrl}) {
```

Comment:
So what is a provider and what is a provisioner and why are they different?

Reply:
oh, ok, provisioner runs all the time and calls methods of the providers it has

Reply:
To summarize the above two comments: maybe some documentation is needed here 😆

Reply:
I think "there's no such thing as a provisioner anymore" is the most accurate answer.  But we have this property called `provisionerId` that still exists for compatibility.  It now means something like "workerTypeGroup".

Reply:
> there's no such thing as a provisioner anymore

I do see `class Provisioner` in the code tho. I meant to ask to add documentation for it

Reply:
Oh, good point!  I thought you were commenting on the `provisionerId` in this line :)

Reply:
no it was just a random line I chose to make a generic comment on the file :)